### ISSUE TITLE
Hidden States?

### ISSUE THREAD

**Issue by norsedrunkensailor** [lucidrains/nGPT-pytorch#12]
Love love this repo! 

For the sake of 'concept alignment' and things like this. Would it be possible to extend the repo to output_hidden_states = True / False? 

So that a forward pass could look like 

`loss, hidden_states = model(encoded["input_ids"], return_loss = True, output_hidden_states = True)`. Probably just by modifying the `for attn, ff in self.layers:` loop in nGPT. 

Would be extremely useful:D

Best, 
NDS

**Comment by lucidrains** [comment]
@norsedrunkensailor good idea! should work now like so

```python
import torch
from nGPT_pytorch import nGPT

model = nGPT(
    num_tokens = 256,
    dim = 512,
    depth = 4,
    attn_norm_qk = True
)

x = torch.randint(0, 256, (2, 2048))

loss, hiddens = model(x, return_loss = True, return_hiddens = True)
loss.backward()

print(hiddens.shape)
```

**Comment by norsedrunkensailor** [comment]
Faster than the wind, you are.
Sbatching jobs asaaaaap

Best,
NDS
